sql/core/src/test/resources/sql-tests/inputs/pgSQL/aggregates_part3.sql

```diff
@@ -0,0 +1,273 @@
+--          generate_series(1,2) i;
+
+-- select * from agg_view1;
+-- select pg_get_viewdef('agg_view1'::regclass);
+
+-- drop view agg_view1;
+
+-- incorrect DISTINCT usage errors
+
+-- select aggfns(distinct a,b,c order by i)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+-- select aggfns(distinct a,b,c order by a,b+1)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+-- select aggfns(distinct a,b,c order by a,b,i,c)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+-- select aggfns(distinct a,a,c order by a,b)
+--   from (values (1,1,'foo')) v(a,b,c), generate_series(1,2) i;
+
+-- [SPARK-27978] Add built-in Aggregate Functions: string_agg
+-- string_agg tests
+-- select string_agg(a,',') from (values('aaaa'),('bbbb'),('cccc')) g(a);
+-- select string_agg(a,',') from (values('aaaa'),(null),('bbbb'),('cccc')) g(a);
+-- select string_agg(a,'AB') from (values(null),(null),('bbbb'),('cccc')) g(a);
+-- select string_agg(a,',') from (values(null),(null)) g(a);
+
+-- check some implicit casting cases, as per bug #5564
+-- select string_agg(distinct f1, ',' order by f1) from varchar_tbl;  -- ok
+-- select string_agg(distinct f1::text, ',' order by f1) from varchar_tbl;  -- not ok
+-- select string_agg(distinct f1, ',' order by f1::text) from varchar_tbl;  -- not ok
+-- select string_agg(distinct f1::text, ',' order by f1::text) from varchar_tbl;  -- ok
+
+-- Skip these tests because we do not have a bytea type
+-- string_agg bytea tests
+-- create table bytea_test_table(v bytea);
+
+-- select string_agg(v, '') from bytea_test_table;
+
+-- insert into bytea_test_table values(decode('ff','hex'));
+
+-- select string_agg(v, '') from bytea_test_table;
+
+-- insert into bytea_test_table values(decode('aa','hex'));
+
+-- select string_agg(v, '') from bytea_test_table;
+-- select string_agg(v, NULL) from bytea_test_table;
+-- select string_agg(v, decode('ee', 'hex')) from bytea_test_table;
+
+-- drop table bytea_test_table;
+
+-- [SPARK-27986] Support Aggregate Expressions with filter
+-- FILTER tests
+
+-- select min(unique1) filter (where unique1 > 100) from tenk1;
+
+-- select sum(1/ten) filter (where ten > 0) from tenk1;
+
+-- select ten, sum(distinct four) filter (where four::text ~ '123') from onek a
+-- group by ten;
+
+-- select ten, sum(distinct four) filter (where four > 10) from onek a
+-- group by ten
+-- having exists (select 1 from onek b where sum(distinct a.four) = b.four);
+
+-- [SPARK-28682] ANSI SQL: Collation Support
+-- select max(foo COLLATE "C") filter (where (bar collate "POSIX") > '0')
+-- from (values ('a', 'b')) AS v(foo,bar);
+
+-- outer reference in FILTER (PostgreSQL extension)
+select (select count(*)
+        from (values (1)) t0(inner_c))
+from (values (2),(3)) t1(outer_c);
+-- Rewriting to CASE WHEN will hit: Expressions referencing the outer query are not supported outside of WHERE/HAVING clauses
```

Comment:
File a JIRA issue?

Reply:
I think we should revert it to original query. We can not support Aggregate Expressions with filter.